core/src/main/java/com/crawljax/core/CandidateElementExtractor.java

```diff
@@ -340,7 +340,7 @@ private boolean isExternal(String href) {
 	 * @return true if href has the pdf or ps pattern.
 	 */
 	private boolean isFileForDownloading(String href) {
-		final Pattern p = Pattern.compile(".+.pdf|.+.ps|.+.zip|.+.mp3");
+		final Pattern p = Pattern.compile(".+\\.pdf$|.+\\.ps$|.+\\.zip$|.+\\.mp3$");
```

Comment:
- Worth creating a non-capturing group for the extensions to avoid repetitions (i.e. `.+\\.(?:pdf|ps|zip|mp3)$`).
 - This could also be extracted into a class variable to avoid compiling it each time.